apps/confidential/confidential_test.go

```diff
@@ -432,6 +432,78 @@ func TestAcquireTokenByAuthCode(t *testing.T) {
 	}
 }
 
+func TestAcquireTokenByAuthCodeTokenExpiry(t *testing.T) {
```

Comment:
this test passes without the fix

Reply:
Yes, because the fix is for the public client. Confidential client works fine.
Wrote this test for perform the **2** case in the list that Bogdan mentioned.